resources/metadata/protecode.yaml

```diff
@@ -0,0 +1,229 @@
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: false
+    - name: addSideBarLink
+      type: bool
+      description: Whether to create a side bar link pointing to the report produced by Protecode or not
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: true
+    - name: verbose
+      type: bool
+      description: Whether to log verbose information or not
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: false
+    - name: protecodeTimeoutMinutes
+      type: string
+      description: The timeout to wait for the scan to finish
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: 60
+    - name: protecodeServerUrl
+      type: string
+      description: The URL to the Protecode backend
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: https://protecode.c.eu-de-2.cloud.sap
+    - name: reportFileName
+      type: string
+      description: The file name of the report to be created
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: protecode_report.pdf
+    - name: useCallback
+      type: bool
+      description: Whether to the Protecode backend's callback or poll for results
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: false
+    - name: fetchUrl
+      type: string
+      description: The URL to fetch the file to scan with Protecode which must be accessible via public HTTP GET request
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: protecodeGroup
+      type: string
+      description: The Protecode group ID of your team
+      mandatory: true
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: null
+    - name: reuseExisting
+      type: bool
+      description: Whether to reuse an existing product instead of creating a new one
+      mandatory: false
+      scope:
+      - GENERAL
+      - PARAMETERS
+      - STAGES
+      - STEPS
+      default: false
+    - name: user
+      type: string
+      description: user which is used for the protecode scan
+      mandatory: true
+      scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+    - name: password
+      type: string
+      description: password which is used for the user
+      mandatory: true
+      scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+    secrets:
+    - name: protecodeCredentialsId
+      type: jenkins
+    - name: dockerCredentialsId
+      type: jenkins
+    resources:
```

Comment:
we don't need this `resources` part in the `input` section.
It is sufficient to refer to the corresponding `commonPipelineEnvironment` parameter names in the step parameters.
This mapping seems to be missing so far ...